.github/workflows/qns.yml

```diff
@@ -122,29 +123,21 @@ jobs:
     steps:
       - uses: actions/download-artifact@018cc2cf5baa6db3ef3c5f8a56943fffe632ef53 # v6.0.0
         with:
-          pattern: '*pr*'
-          path: /tmp
-
-      - uses: actions/download-artifact@018cc2cf5baa6db3ef3c5f8a56943fffe632ef53 # v6.0.0
-        if: ${{ github.event_name == 'pull_request' }}
-        with:
-          pattern: '*main*'
+          pattern: '* Docker image'
           path: /tmp
+          merge-multiple: true
 
       - run: |
-          docker load --input "/tmp/pr.tar"
+          docker load --input /tmp/pr.tar
           PR_IMAGE=$(docker images --format "{{.Repository}}:{{.Tag}}" | grep pr | head -1)
```

Comment:
Aren't we naming the image, which would let this avoid grep here?  Or don't you want to include `ghcr.io/${{ github.repository }}-qns-{main|pr}` here? I'd have thought that preferable to `grep|head`.